### PR TITLE
Leaderboard - API

### PR DESCRIPTION
**Pull request by jmontanholi**
## In this branch I implemented the following: 

* ### Request.js
   * #### createGame function to create the game inside the API
   * #### getScores function to fetch the leaderboard scores from my game
   * #### createScores function to create scores and store them inside my game leaderboard
* ### Refactor of populate function to use the API list to create the DOM
* ### Refactor of submit button functionality to add items to the API list
* ### Refresh button functionality to populate again after you insert an item

### CODE REVIEW

File: src/request.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    body: JSON.stringify({
+      name: 'Montanholi\'s Game',
+    }),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  })
+    .then((response) => response.json())
+    .then((json) => json.result.split(' ')[3]);
+};
```
Comment (DeliceLydia) [inline src/request.js]: - [x] Please Use `async` and `await` with the fetch method to send and receive data.

File: src/request.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+const getScores = () => {
+  const scores = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/oqptEhnYZtDs1PlRT1tV/scores';
+  return fetch(scores)
+    .then((response) => response.json())
+    .then((json) => json.result);
+};
+
```
Comment (DeliceLydia) [inline src/request.js]: - [x] Use `async` and `await` with the fetch method to send and receive data.

File: src/request.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      user,
+      score,
+    }),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  })
+    .then((response) => response.json())
+    .then((json) => json.result);
+};
```
Comment (DeliceLydia) [inline src/request.js]: - [x] Use `async` and `await` with the fetch method to send and receive data.

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @jmontanholi, 

Good job so far! there are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._


### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/request.js [hunk 4]
```diff
@@ -1,28 +1,28 @@
...
       'Content-type': 'application/json; charset=UTF-8',
     },
-  })
-    .then((response) => response.json())
-    .then((json) => json.result.split(' ')[3]);
-};
+  });
+  const result = await response.json();
+  return result.result.split(' ')[3];
+}
```
Comment (DeliceLydia) [inline src/request.js]: - [x] Good job I can see you used `await` and `async` but according to the requirements avoid using `function` keyword but instead use `arrow function`

File: src/request.js [hunk 5]
```diff
@@ -1,28 +1,28 @@
...
-  return fetch(scores)
-    .then((response) => response.json())
-    .then((json) => json.result);
-};
+async function getScores() {
+  const scores = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/7MRZkAe2WgnNaEn7ujCD/scores';
+  const response = await fetch(scores);
+  const result = await response.json();
+  return result.result;
+}
```
Comment (DeliceLydia) [inline src/request.js]: - [x] Good job I can see you used `await` and `async` but according to the requirements avoid using `function` keyword but instead use `arrow function`

File: src/request.js [hunk 6]
```diff
@@ -1,28 +1,28 @@
...
 
-const createScores = (user, score) => {
-  const scores = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/oqptEhnYZtDs1PlRT1tV/scores';
-  return fetch(scores, {
+async function createScores(user, score) {
+  const scores = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/7MRZkAe2WgnNaEn7ujCD/scores';
+  const response = await fetch(scores, {
     method: 'POST',
     body: JSON.stringify({
       user,
```
Comment (DeliceLydia) [inline src/request.js]: - [x] Good job I can see you used `await` and `async` but according to the requirements avoid using `function` keyword but instead use `arrow function`

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @jmontanholi, 

Great work on making the changes requested previously 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by carlos-ssh — APPROVED** [review]
Hi @jmontanholi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**